server/server.go

```diff
@@ -3501,15 +3504,22 @@ func (s *Server) String() string {
 	return s.info.Name
 }
 
-func (s *Server) startGoRoutine(f func()) bool {
+func (s *Server) startGoRoutine(f func(), labels ...string) bool {
 	var started bool
 	s.grMu.Lock()
+	defer s.grMu.Unlock()
 	if s.grRunning {
 		s.grWG.Add(1)
-		go f()
+		go func(applyLabels bool) {
```

Comment:
Wonder if we make this a simple server call that can be called from any go routine as needed to setup labeling vs as part of startGoRoutine?

Reply:
It can be done either way yep. Doing it in `startGoRoutine` just seemed like it had the added benefit of giving context to the reader at the point of invocation, almost self-documenting.